pkg/operation/botanist/controlplane.go

```diff
@@ -304,6 +297,46 @@ func (b *Botanist) HibernateControlPlane(ctx context.Context) error {
 	return nil
 }
 
+// PrepareControlPlaneDeployments scales the kube-apiserver, kube-controller-manager and gardener-resource-manager to zero and deletes the vpa for the kube-apiserver
+func (b *Botanist) PrepareControlPlaneDeployments(ctx context.Context) error {
+	if err := b.scaleControlPlaneDeploymentsToZero(ctx, b.K8sSeedClient.Client()); err != nil {
+		return err
+	}
+
+	if err := b.K8sSeedClient.Client().Delete(ctx, &hvpav1alpha1.Hvpa{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.DeploymentNameKubeAPIServer, Namespace: b.Shoot.SeedNamespace}}, kubernetes.DefaultDeleteOptions...); err != nil {
+		if !apierrors.IsNotFound(err) && !meta.IsNoMatchError(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ScaleETCDStatefulSetsToZero scales ETCD main and events to zero
+func (b *Botanist) ScaleETCDStatefulSetsToZero(ctx context.Context) error {
+	for _, statefulset := range []string{v1beta1constants.ETCDEvents, v1beta1constants.ETCDMain} {
+		if err := kubernetes.ScaleStatefulSet(ctx, b.K8sSeedClient.Client(), kutil.Key(b.Shoot.SeedNamespace, statefulset), 0); client.IgnoreNotFound(err) != nil {
```

Comment:
Instead of scaling down the statefulset, now that #1762 is merged, you should set the [`replicas`](https://github.com/gardener/gardener/blob/master/charts/seed-controlplane/charts/etcd/templates/etcd.yaml#L70) field to `0`.

Reply:
> I am wondering how the gardenlet in the source seed will get to know that it has to start the migration preparation flow. If I understood correctly, the trigger is to change `.spec.seedName` from `old` to `new`, and then the gardenlet responsible for `old` should start the migration prep flow, right? However, [this event filtering](https://github.com/gardener/gardener/blob/master/pkg/gardenlet/controller/shoot/shoot.go#L120) will prevent that it enqueues the shoot, hence, nothing will happen. Also, I don't see any code related to preventing the gardenlet responsible for `new` to wait until the preparation is finished, or?

You are right. I can change the FilterFunc to pass the Shoot if the Gardenlet in the source Seed WAS responsible meaning has the seed as status,  but to skip it in the Target Gardenlet if it has different status.SeedName not equal to the SeedName form the Config . And in the shoot control if it was responsible to prepare it.

Reply:
`return *shoot.Spec.SeedName == seedName && (shoot.Status.SeedName == nil || *shoot.Status.SeedName == seedName) || (shoot.Status.SeedName != nil && *shoot.Status.SeedName == seedName)
`